onnxruntime/core/providers/cpu/nn/pool.cc

```diff
@@ -270,14 +272,15 @@ Status Pool<float, MaxPool<8 /*VERSION*/>>::Compute(OpKernelContext* context) co
         int64_t* i_d = I_data ? I_data + c * y_step : nullptr;
         for (int64_t ph = 0; ph < pooled_height; ++ph) {
           int64_t hstart = ph * stride_h() - pads[0];
-          int64_t hend = std::min(hstart + kernel_shape[0] * dilation_h - dilation_h + 1, height);
-          hstart = std::max(hstart, static_cast<int64_t>(0));
+          int64_t hend = hstart + kernel_shape[0] * dilation_h;
```

Comment:
Why not just replace

```c
          hstart = std::max(hstart, static_cast<int64_t>(0));
 ```
with the snippet below?
```c
          while (hstart < 0) hstart += dilation_h;
```

...and similarly for **dstart** and **wstart** below? That passes the new unit tests and doesn't require the extra function **math::is_a_ge_zero_and_a_lt_b()**.